MetaMask/metamask-mobile
# [FIX] [IMPROVEMENT] keep nft when transaction fail

**Description**
The problem was the collectible was removed when the transaction started.

**Proposed Solution**
The solution proposed was that the collectible was only removed when the transaction was confirmed.
Also, it's proposed here when the transaction it's submitted and we are waiting for feedback from the blockchain, the collectible details will change the _Send_ button for a _View Transactions_ button, which will redirect the user to the Activity View option on the left menu, where he can see the collectible transaction and cancel it or speed it up.

Also sorted the collectible by it's name inside of the collection.

**Code Impact**
Medium, changes the collectible details component and the way that collectibles have in the wallet.

**Screenshots/Recordings**
Small demo of the solution:
UPDATE:  https://recordit.co/6b8oh8xWLG
**Test Cases**
Case1:
 - Import collectible
 - Transact collectible to another account
 - Press the collectible to see the collectible details
 - Click on View Transaction Button, will redirect to Activity View
 - Cancel the collectible transaction 
 - Go to wallet
 - Verify if can send the collectible again
Case2:
 - Import collectible
 - Transact collectible to another account
 - Press the collectible to see the collectible details
 - Wait for the transaction ends and touch the modal to the button update

**Checklist**


**https://github.com/MetaMask/metamask-mobile/issues/2784#issuecomment-1038345711**

Solves #2784
